cdap-api/src/main/java/io/cdap/cdap/api/app/ApplicationUpgradeContext.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.app;
+
+import io.cdap.cdap.api.artifact.ArtifactId;
+
+import javax.annotation.Nullable;
+
+/**
+ * Context for upgrading Application configs. Provides helper methods for application to supporting config upgrade.
+ */
+public interface ApplicationUpgradeContext {
+
+  /**
+   * Gets latest available artifact information for a given plugin type and name.
+   *
+   * @param pluginType the plugin type
+   * @param pluginName the plugin name
+   * @return artifact information for the plugin type and name.
+   *         returns null if no artifact for the plugin could be found.
+   */
+  @Nullable
+  ArtifactId getLatestPluginArtifact(String pluginType, String pluginName);
```

Comment:
don't we have use cases where we want to ignore the scope when getting the 'latest' artifact? This API looks like it won't be enough for all use cases.

Reply:
Added scope as another argument based on the discussion.